utils/build-script-impl

```diff
@@ -2089,6 +2089,16 @@ for host in "${ALL_HOSTS[@]}"; do
                     -Ddispatch_DIR:PATH=$(build_directory ${host} libdispatch)/cmake/modules
                     -DFoundation_DIR:PATH=$(build_directory ${host} foundation)/cmake/modules
                 )
+
+                # Ensure on Darwin platforms that we prefer the SQLite from the SDK
+                # instead of picking one found elsewhere (e.g. in /usr/include )
+                if [[ "$(uname -s)" = "Darwin" ]]; then
+                    cmake_options=(
+                        "${cmake_options[@]}"
+
+                        -DSQLite3_INCLUDE_DIR:PATH=$(xcrun -sdk macosx -show-sdk-path)/usr/include
```

Comment:
You should also specify `-DSQLite3_LIBRARY` to ensure that the associated library is linked.

Reply:
That seems not needed -- for my test trials on macOS 10.14.4, the SQLite3_LIBRARY seems populated correctly (in this case it's always`/usr/lib/libsqlite3.dylib` and the SDK only contains the headers)

Reply:
Sure, but I could also see the headers also being properly populated as well.  The point is that this would be completely explicit as to which library and header set should be hard.  This is being explicit to ensure that we don’t happen to get a mismatched set of library and headers.

Reply:
Incorporated feedback in 7f4d790c169a16728ad198113bd0cdea85e01e2c